File: frontend/src/domain/entities/Factura_entity.ts
import { TimeStamp } from "../values/Timestamp";

export class FacturaEntity extends TimeStamp {
    id: number | null;
    venta: number;
    fecha_expendida: string | null;
    nit: string | null;

    constructor(id: number | null, venta: number, fecha_expendida: string | null, nit: string | null,
        created_at: Date | null = null,
        updated_at: Date | null = null,
        deleted_at: Date | null = null
    ) {
        super(created_at, updated_at, deleted_at);
        this.id = id;
        this.venta = venta;
        this.fecha_expendida = fecha_expendida;
        this.nit = nit;
    }
}

File: backend/app_api/domain/entities/Factura_entity.py
from dataclasses import dataclass
from datetime import date
from app_api.domain.values.timestamp import TimestampModel

@dataclass(kw_only=True)
class FacturaEntity(TimestampModel):
    id: int | None
    venta: int
    fecha_expendida: date | None
    nit: str | None
